### PR TITLE
ACP 3507 - ACP 3636 Add and update documentation for headless approach and APIs

### PR DESCRIPTION
**Pull request by stereomon**
## PR Description

- Added description for headless approach.
- Added new page with all sync endpoints and all async messages.

TODO add diagrams.

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+* `PaymentCaptureFailedTransfer`: Send from the App when the payment capture fails.
+* `PaymentRefundedTransfer`: Send from the App when the payment is refunded.
+* `PaymentRefundFailedTransfer`: Send from the App when the payment refund fails.
+* `PaymentCanceledTransfer`: Send from the App when the payment is canceled.
+* `PaymentCancellationFailedTransfer`: Send from the App when the payment cancellation fails.
+* `PaymentCreatedTransfer`: Send from the App when the payment is created.
+* `PaymentUpdatedTransfer`: Send from the App when the payment is updated.
+* `ReadyForMerchantAppOnboardingTransfer`: Send from the App when the App is ready to onboard merchants.
+* `MerchantAppOnboardingStatusChangedTransfer`: Send from the App when the merchant app onboarding status changes.
+* `AppConfigUpdatedTransfer`: Send from the App when the App configuration is updated.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:40]: All of the messages are without Transfer suffix

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Synchronous API
+
+All Payment Service Provider (PSP) integrations are based on the synchronous API. The synchronous API is used to process payments and refunds. The following endpoints are used:
+
++ `/configure`: Used from the AppStoreCatalog to configure an App.
++ `/disconnect`: Used from the AppStoreCatalog to disconnect an App.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:47]:
```suggestion
+ `/configure`: Used from the ACP App Catalog to configure an App.
+ `/disconnect`: Used from the ACP App Catalog to disconnect an App.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
++ `/initialize-payment`: Used from the Back office after an order was created and before the hosted payment page is shown to the customer. This will initialize the payment in the PSP App and the PSP App will return the URL to the hosted payment page.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:48]:
```suggestion
+ `/initialize-payment`: Used from the SCOS back-end after an order was created and before the hosted payment page is shown to the customer. This will initialize the payment in the PSP App and the PSP App will return the URL to the hosted payment page.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
++ `/confirm-pre-order-payment`: Used from the Back office after an order was created in a headless approach where the payment gets created before the order is persisted. This will connect a previously created (preOrder) Payment on App side with the order on Zed side.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:49]:
```suggestion
+ `/confirm-pre-order-payment`: Used from the SCOS back-end after an order was created in a headless approach where the payment gets created before the order persists. This will connect a previously created (preOrder) Payment on the App side with the order on the Zed side.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
++ `/cancel-pre-order-payment`: Used from the Glue application in a headless approach where a customer clicks cancel or in cases where the headless implementation sees the need for canceling. This will cancel the Payment on the PSP side.
++ `/payments/transfers`: Used from the Back office with the App being used in a Marketplace business model. This initiates the transfer of money from the Marketplace to the Merchant.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:51]:
```suggestion
+ `/payments/transfers`: Used from the Back office/OMS with the App being used in a Marketplace business model. This initiates the transfer of money from the Marketplace to the Merchant.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 6]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
 
 ## Checkout payment step
 
+Implementing the Payment Provider Stripe into your project can be done in many different ways.
+
+- You can add it in your own headless frontend application using Glue.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:92]:
```suggestion
- You can add it in your headless frontend application using the Glue application.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 7]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
...
+
+[//]: # (- You can use the default implementation in the Payment selection page using Yves which shows the Payment Elements then later on the summary page.)
+- You can add it as a hosted Payment page which uses a redirect after your customer submits the order.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:95]:
```suggestion
- You can add it as a hosted payment page which uses a redirect after your customer submits the order.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 8]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
...
+#### Modules
+
+- `spryker/kernel-app:1.2.0`
+- `spryker/payment:5.24.0`
+- `spryker/payments-rest-api:1.3.0`
+
+#### PreOrder payment flow in a Nutshell
+
+- The customer either selects Stripe as the payment method or he gets only Stripe Elements presented.
+- When you have more than one Payment Provider in your project and Stripe gets selected the `InitializePreOrderPayment` Glue API endpoint (glue.your-website.com/payments?action=initialize-pre-order-payment) is called with the Payment Provider name (Stripe), the Payment Method name (Stripe), CSS-ID of the element to render Stripe Elements to, CSS-ID of the element to render messages to, the Payment Element Options including the `return_url`, and the Quote data. See example below.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:110]:
```suggestion
- When you have more than one Payment Provider in your project and Stripe gets selected the `InitializePreOrderPayment` Glue API endpoint (glue.your-website.com/payments) is called with the Payment Provider name (Stripe), the Payment Method name (Stripe), payment amount, and the Quote data. See example below.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 9]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
...
+- When you only use Stripe in your project then the `InitializePreOrderPayment` Glue API endpoint (glue.your-website.com/payments) is called with the Payment Provider name (Stripe) and the Payment Method name (Stripe). See example below.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:111]: quote and payment amount is still needed, so there is no difference with the previous paragraph, should be combined into one

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 10]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
...
+  - The required quote data must be provided by your application.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:112]:
```suggestion
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 11]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
...
+- Zed now makes the API call to the Stripe App including required authorization.
+- On the Stripe App side the Payment with the given data is persisted and an API call to Stripe is made to get the ClientSecret and the PublishableKey.
+- You will get back a JSON response with the ClientSecret and the PublishableKey.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:115]: and PaymentIntent ID + AccountID for direct B2C/B2B model

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 12]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
...
+- Use the example JavaScript to render the Stripe Elements on the summary page of your application.
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:116]:
```suggestion
- Use the example JavaScript to render the Stripe Elements on the order summary page of your application.
```

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
++ `/webhooks`: Used from external applications to send requests to an App.
++ `/webhooks/test`: Used from external applications in test mode to send requests to an App.
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:53]: Missing endpoints:
- /payments
- /payment-cancellations

Will be added after moving.

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+
+---
+
+This document provides an overview of the APIs. All Payment Service Provider (PSP) use a synchronous API together with an asynchronous API.
+
+## Asynchronous API
+
+All Payment Service Provider (PSP) integrations are based on the asynchronous API. The asynchronous API is used to process payments and refunds. The following asynchronous messages are used:
+
+* `AddPaymentMethodTransfer`: Send from the App when a new payment method is added.
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:22]: @matweew should I remove the Transfer suffix?

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 15]
```diff
@@ -87,6 +87,305 @@ For more information about ACP payment methods integration with your project OMS
...
+- Then the customer can select the Payment Method in the Stripe Elements and submits the data.
+- The customer will then be redirected to the provided `return_url` which must make another Glue request (glue.your-website.com/checkout) to persist the order in the backoffice.
+- After this the customer should be redirected to the success page of your application.
```
[the remark this answers is not in the document]
Comment (stereomon) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:118]: It's in the making.

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+   - /docs/pbc/all/payment-service-provider/202311.0/base-shop/third-party-integrations/stripe/install-stripe.html
+   - /docs/pbc/all/payment-service-provider/202311.0/base-shop/third-party-integrations/stripe/integrate-stripe.html
+
+---
+
+This document provides an overview of the APIs. All Payment Service Provider (PSP) use a synchronous API together with an asynchronous API.
+
+The following diagram explains the Hosted Payment Page Flow. 
+
+PLACE OVERVIEW IMAGE FOR HOSTED PAYMENT PAGE HERE
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:20]: HostedPaymentPageFlow

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+
+You can find information about endpoints and messages down below.
+
+The following diagram explains the Headless Payment Page Flow using Glue.
+
+PLACE OVERVIEW IMAGE FOR HEADLESS PAYMENT PAGE HERE
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:26]: HeadlessPaymentFlow

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+
+The following diagram explains the configuration and disconnect flow for a Payment App.
+
+PLACE OVERVIEW IMAGE FOR CONFIGURATION/DISCONNECTION FLOW HERE
+
+The optional elements here are the `AddPaymentMethod`, `UpdatePaymentMethod`, `DeletePaymentMethod`, and the `ReadyForMerchantAppOnboarding` messages. The Payment Method related messages are used to manage payment methods in the SCOS. Those messages are only sent when a Payment method configuration changes or when the available payment methods are changed. The `ReadyForMerchantAppOnboarding` message is used to inform SCOS that the App is ready to onboard merchants, this message is send when the App is used in the Marketplace business model.
+
+The following diagram explains the flow of an Order in the OMS together with an App.
+
+PLACE OVERVIEW IMAGE FOR OMS FLOW HERE
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:26]: OMSPaymentFlow

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+
+The optional elements here are the request to do transfers `/payments/transfers` which is only used in a Marketplace business model, the `CancelPayment` message which is only used when a payment needs to be canceled, and the `RefundPayment` message which is only used when the refund process is triggered for one or more order items.
+
+The following diagram explains the Hosted Payment Page Flow. 
+
+PLACE OVERVIEW IMAGE FOR HOSTED PAYMENT PAGE HERE
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:32]: HostedPaymentPageFlow

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+
+The following diagram explains the Headless Payment Page Flow using Glue.
+
+PLACE OVERVIEW IMAGE FOR HEADLESS PAYMENT PAGE HERE
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/apis-and-overview-diagrams.md:36]: HeadlessPaymentFlow

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 21]
```diff
@@ -87,6 +87,307 @@ For more information about ACP payment methods integration with your project OMS
...
+            paymentProviderName: 'stripe',  // taken from /checkout-data?include=payment-methods
+          },
+          preOrderPaymentData: {
+            "transactionId": this.transactionId, // This is empty in the first request but has to be used in further requests 
+          },
+        },
+      },
+    };
+
+    const responseData = await this.fetchHandler(`GLUE_APPLICATION_URL/payments`, {
```
Comment (matweew) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:153]: To make it usable for projects, they must also configure CORS correctly for Glue application. We have a guide for this.
https://docs.spryker.com/docs/pbc/all/miscellaneous/202404.0/install-and-upgrade/install-glue-api/install-the-spryker-core-glue-api.html#configure-cors

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 22]
```diff
@@ -80,13 +83,340 @@ This example demonstrates how to configure the order state machine transition fr
...
+
+#### PreOrder payment flow
+
+1. The customer either selects Stripe as the payment method or Stripe Elements is loaded by default.
+2. When Stripe is selected, the `InitializePreOrderPayment` Glue API endpoint (`glue.mysprykershop.com/payments`) is called with the following parameters:
+  * Payment provider name: Stripe
+  * Payment method name: Stripe
+  * Payment amount
+  * Quote data
+3. Zed makes the API call to the Stripe App, including required authorization.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:124]: Back Office?

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 23]
```diff
@@ -80,13 +83,341 @@ This example demonstrates how to configure the order state machine transition fr
...
+4. The payment with the given data is persisted On the Stripe App side.
+5. An API call to Stripe is made to get `ClientSecret` and `PublishableKey` keys.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:126]: Does the Stripe App make the call?

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 24]
```diff
@@ -80,13 +83,341 @@ This example demonstrates how to configure the order state machine transition fr
...
+5. Stripe returns a JSON response with the following parameters:
+  * TransactionId
+  * ClientSecret
+  * PublishableKey
+  * Only for marketplaces: AccountId
+- Use the example JavaScript to render Stripe Elements on the order summary page.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md]: How is this information related to this flow?

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 25]
```diff
@@ -80,13 +83,341 @@ This example demonstrates how to configure the order state machine transition fr
...
+6. The customer selects a payment method in Stripe Elements and submits the data.
+7. The customer is redirected to the provided `return_url`, which makes another Glue API request (`glue.mysprykershop.com/checkout`) to persist the order in the Back Office.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md]: Provided by who and where?

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 26]
```diff
@@ -80,13 +83,342 @@ This example demonstrates how to configure the order state machine transition fr
...
+8. The customer is redirected to the success page of your application.
+9. Through the `\Spryker\Zed\Payment\Communication\Plugin\Checkout\PaymentConfirmPreOrderPaymentCheckoutPostSavePlugin` plugin, the PreOrder payment is confirmed on the Stripe App side. The `order_reference` is passed to the Stripe app to be connected with `transaction_id`.
+10. After the payment is processed on the Stripe App side, a `PaymentUpdated` message is sent to Spryker.
+  The message contains additional data, which you can see in the Back Office.
+11. Depending on payment status, one of the following messages is returned through an asynchronous API, which moves the order to the next state in OMS:
+  *  When the payment is successful, a `PaymentAuthorized` message is returned.
+  *  When the payment is failed, a `PaymentAuthorizationFailed` message is returned.
+12. Further on, the order is processed through the OMS.
+
+
```
Comment (andriitserkovnyi) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:144]: From this point on, the information needs to be restructured based on what exactly we are trying to explain

**Review by andriitserkovnyi — CHANGES_REQUESTED** [review]

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/install-and-configure-stripe-prerequisites.md [hunk 27]
```diff
@@ -262,6 +268,33 @@ use SprykerShop\Yves\PaymentPage\Plugin\PaymentPage\PaymentForeignPaymentCollect
...
+            new PaymentsResourceRoutePlugin(),
+            new PaymentCancellationsResourceRoutePlugin(),
+        ];
+    }
+
+```
+
+## Enable CORS
+
+When using the headless approach it is required to configure CORS for the Glue application. YOu can get more informations about this in the [CORS configuration](/docs/pbc/all/miscellaneous/202404.0/install-and-upgrade/install-glue-api/install-the-spryker-core-glue-api.html#configure-cors) article.
```
Comment (stereomon) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/install-and-configure-stripe-prerequisites.md:297]: @andriitserkovnyi please check if the link is correct.

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md (hunk 21 above)
Comment (stereomon) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:173]: Added here https://github.com/spryker/spryker-docs/pull/2837/files#diff-90038c83c8469fb56176e83bfc5f20add6232b7a7769acb30d9900b4a3789a89R295

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md (hunk 26 above)
Comment (stereomon) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:145]: Is it better now?

Comment (andriitserkovnyi) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:146]: Nope, I still don't understand the idea behind this section. Perhaps break it down into smaller sections

Comment (stereomon) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:146]: The above section describes the PreOrder Payment flow and the below sections describe where to find the example integration as well as the examples with description.

I don't see how to change it to be better.

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 28]
```diff
@@ -80,13 +83,346 @@ This example demonstrates how to configure the order state machine transition fr
...
+### Example integration
+
+To check out how the integration works, see this [example application](https://github.com/spryker-projects/spa-checkout-glue-with-stripe).
+
+Before the customer is redirected to the summary page, all required data is collected: customer data, addresses, and selected shipment method. When the customer goes to the summary page, to get the data required for rendering the Stripe Elements, the application needs to call the `InitializePreOrderPayment` Glue API endpoint.
+
+#### Example of rendering Stripe Elements on the summary page
+
+```JS
```
Comment (andriitserkovnyi) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:155]: I assume this code is the example of rendering stripe elements on the summary page?

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 29]
```diff
@@ -80,13 +83,346 @@ This example demonstrates how to configure the order state machine transition fr
...
+
+    this.transactionId = paymentProviderData.transactionId;
+    this.accountId = paymentProviderData.accountId; // only be used on the Direct business model. When using a Marketplace business model this will not be present.
+
+    await this.setupStripe();
+  }
+
+```
+
+To identify the customer, you can use the `Authorization` and `X-Anonymous-Customer-Unique-Id` headers.
```
Comment (andriitserkovnyi) [inline docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:195]: Why do I need to identify the customers? How is it related to the code above?

Comment (stereomon) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:195]: This is usual in Glue checkouts. Either you have a logged-in customer or a guest customer.

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md [hunk 30]
```diff
@@ -80,13 +83,346 @@ This example demonstrates how to configure the order state machine transition fr
...
+  *  Payment is failed: `PaymentAuthorizationFailed` message.
+14. Further on, the order is processed through the OMS.
+
+All payment related messages mentioned above are handled by `\Spryker\Zed\Payment\Communication\Plugin\MessageBroker\PaymentOperationsMessageHandlerPlugin`, which is registered in `MessageBrokerDependencyProvider`.
+
+### Example integration
+
+To check out how the integration works, see this [example application](https://github.com/spryker-projects/spa-checkout-glue-with-stripe).
+
+Before the customer is redirected to the summary page, all required data is collected: customer data, addresses, and selected shipment method. When the customer goes to the summary page, to get the data required for rendering the Stripe Elements, the application needs to call the `InitializePreOrderPayment` Glue API endpoint.
```
[the remark this answers is not in the document]
Comment (stereomon) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:154]: As it is a general requirement for a Glue checkout.

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md (hunk 28 above)
Comment (stereomon) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:158]: No this is the initial request to get the required data from Stripe to render the payment elements. This includes the clientSecret, the publishableKey, and the accountId.

File: docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md (hunk 30 above)
[the remark this answers is not in the document]
Comment (andriitserkovnyi) [reply docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md:151]: Can we rename the title for this paragraph then? The structure is different now
